charts/linkerd2/templates/controller.yaml

```diff
@@ -90,6 +90,12 @@ spec:
       {{- include "linkerd.pod-affinity" $local | nindent 6 -}}
       {{- end }}
       containers:
+      {{- if not (empty .Values.publicAPIProxyResources) }}
+      {{- $r := merge .Values.publicAPIProxyResources .Values.proxy.resources }}
+      {{- $_ := set $tree.Values.proxy "resources" $r }}
+      {{- end }}
+      {{- $_ := set $tree.Values.proxy "await" true }}
```

Comment:
Because core components are not admitted by the proxy-injector, we cannot rely on annotating these components with `config.linkerd.io/proxy-await: "enabled"`.

Therefore, the template must override `Values` here so that `proxy.await` is always `true`. This ensures that when users install Linkerd and explicitly disable this feature for their application, the control plane still has the feature enabled.

This is true for `templates/{destination.yaml, proxy-injector.yaml, sp-validator.yaml}`.